src/emoji/Emoji.h

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include <QObject>
+#include <array>
+
+namespace emoji {
+Q_NAMESPACE
```

Comment:
You probably want a cpp file, that includes the moc file for this, so that it matches the other files.